students/lisa_ferrier/lesson04/mailroom_pt2.py

```diff
@@ -0,0 +1,178 @@
+            ''')
+        donor_name = donor_name.title()
+
+        if donor_name == 'Home':
+            print('\nReturning to main menu...\n')
+            main_menu()
+        elif donor_name == 'List':
+            print(list(donors))
+        else:
+            for name, donations in donors.items():
+                if name in donor_name:
+                    send_thank_yous(donor_name)
+            else:
+                print('\nDonor not found in donor list. Add donation?')
+                response = input('Enter yes or no\n')
+                response = response.lower()
+                while True:
+                    if response == 'yes':
+                        add_donation(donor_name)
+                    elif response == 'no':
+                        verify_donor_info()
+                    else:
+                        print('Please enter yes or no.\n')
+    main_menu()
+
+
+def add_donation(donor_name):
+    '''
+    Adds a donation to the database
+    Inputs:
+        donor_name (function argument)
+        donation_amount (function prompt)
+    After donation is entered, generates a thank you letter to donor.
+    '''
+    while True:
+        try:
+            donation_amt = float(input('\nEnter a donation amount\n'))
+        except ValueError:
+            print('\nPlease enter a dollar amount.\n')
+        else:
+            if donors.get(donor_name):
+                donors[donor_name].append(donation_amt)
+            else:
+                donors[donor_name] = [donation_amt]
+        send_thank_yous(donor_name)
+
+
+def exit_program():
+    ''' Close out of program '''
+    sys.exit()
+
+
+def menu_switch(argument):
+    '''
+    Main menu switch used to control program flow.
+    '''
+    switcher = {
+        1: verify_donor_info,
+        2: gather_all_donors,
+        3: create_report,
+        4: exit_program,
+    }
+    # Get the function from switcher dictionary
+    func = switcher.get(argument)
+    # Execute the function
+    return func()
+
+
+def main_menu():
+    '''
+    Main menu executes on program launch. Asks the user to select a menu option and action is implemented using the menu_switch function to complete specified action.
+    '''
+    print('''
+    Main Menu, select an option:
+    [1] - Thank individual donor.
+    [2] - Thank all donors.
+    [3] - Create a report summarizing donations.
+    [4] - Quit program.
+    ''')
+
+    while True:
+        try:
+            selection = int(input("Please select an option from above (1-4)\n"))
+        except ValueError:
+            print('''\nSorry, I didn't understand that. Please enter an integer
+            between 1 and 4.\n''')
+            # better try again... Return to the start of the loop
+            continue
+        else:
+            if selection > 0 and selection <= 4:
+                # implement switch
+                menu_switch(selection)
+                break
```

Comment:
you don't want to have a `break` here, per the tutorial below this should be a continuing loop


https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom_tutorial.html#controlling-main-program-flow